src/components/Molecules/CtaPrompt/CtaPrompt.component.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * @file CtaPrompt.component.js
+ * Exports a CTA Prompt component.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import CtaMessage from '../CtaMessage/CtaMessage.component';
+
+const cookiePrefix = 'cta_prompt_';
+
+/**
+ * Component that renders a CTA (Call to action) prompt.
+ */
+export default class CtaPrompt extends Component {
+  static propTypes = {
+    children: PropTypes.node,
+    type: CtaMessage.propTypes.type
+  };
+
+  static defaultProps = {
+    children: [],
+    type: null
+  };
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      shownMessage: null
+    };
+
+    if (props.children) {
+      const children = (props.children.length
+        ? props.children
+        : [props.children]
+      ).filter(child => child.type === CtaMessage);
+
+      // Determine which (if any) message should be shown.
+      children.forEach(msg => {
+        const { data } = msg.props;
+
+        const hash = this.getCookie(data.name);
+
+        // Show first message that either:
+        // - Doesn't have a cookie.
+        // - Cookie value doesn't match message hash.
+        if (!this.state.shownMessage && (!hash || hash !== data.hash)) {
+          this.state.shownMessage = msg;
+        }
+      });
+    }
+  }
+
+  getCookie = name => {
```

Comment:
For interacting with cookies, I'd consider potentially using something like this: https://www.npmjs.com/package/react-cookies

It abstracts a lot of the complicated logic away and provides a neat API that makes it easy for you to sync a state property with a cookie.